detect_bitbucket.go

```diff
@@ -1,66 +0,0 @@
-package getter
-
-import (
-	"encoding/json"
-	"fmt"
-	"net/http"
-	"net/url"
-	"strings"
-)
-
-// BitBucketDetector implements Detector to detect BitBucket URLs and turn
-// them into URLs that the Git or Hg Getter can understand.
-type BitBucketDetector struct{}
```

Comment:
Here I was wondering if we should may be make the git getter have the ability to have multiple detectors like this one. For example:

```go
GitGetter{
	Detectors: []{BitBucketDetector, GithubDetector, ...}
}
```

This would largely reduce the diff of this PR and sort of reduce friction for anyone reviewing this and for git history. Because the current detectors will be sort of kept very similar and in the same place.

🤔 What do you think